### DISCUSSION TITLE
Can the regions plugin delete regions based on their IDs?

### DISCUSSION THREAD

**Discussion by lvhaoyu39** [katspaugh/wavesurfer.js#3125] [Q&A] [answered]
Hello, can regions delete regions based on their IDs?

**Comment by katspaugh** [comment, marked as answer]
Not out of the box but you can simply filter them like this:

```js
wsRegions.getRegions()
  .filter((region) => region.id === idToDelete)
  .forEach((region) => region.remove())
```

**Reply by lvhaoyu39** [reply]
Hello, after deleting a certain region, I used the getRegions() method to obtain a list of all regions, including the previously deleted region data.

**Comment by lvhaoyu39** [comment]
Thank you, this is very useful.
